homeassistant/components/inkbird/__init__.py

```diff
@@ -2,25 +2,55 @@
 
 from __future__ import annotations
 
+from typing import Any
+
 from inkbird_ble import INKBIRDBluetoothDeviceData
 
+from homeassistant.components.bluetooth import async_last_service_info
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
-from homeassistant.core import HomeAssistant
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
 
-from .const import CONF_DEVICE_TYPE
+from .const import CONF_DEVICE_DATA, CONF_DEVICE_TYPE, DOMAIN
 from .coordinator import INKBIRDActiveBluetoothProcessorCoordinator
 
-PLATFORMS: list[Platform] = [Platform.SENSOR]
-
 INKBIRDConfigEntry = ConfigEntry[INKBIRDActiveBluetoothProcessorCoordinator]
 
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: INKBIRDConfigEntry) -> bool:
     """Set up INKBIRD BLE device from a config entry."""
+    assert entry.unique_id is not None
     device_type: str | None = entry.data.get(CONF_DEVICE_TYPE)
-    data = INKBIRDBluetoothDeviceData(device_type)
-    coordinator = INKBIRDActiveBluetoothProcessorCoordinator(hass, entry, data)
+    device_data: dict[str, Any] | None = entry.data.get(CONF_DEVICE_DATA)
+    coordinator = INKBIRDActiveBluetoothProcessorCoordinator(hass, entry)
+
+    @callback
+    def _async_device_data_changed(new_device_data: dict[str, Any]) -> None:
+        """Handle device data changed."""
+        hass.config_entries.async_update_entry(
+            entry, data={**entry.data, CONF_DEVICE_DATA: new_device_data}
+        )
```

Comment:
We need to remember the device data since when the device changes from C to F it will send the data in C or F and  if we don't remember the last config update that told us if the device is in C or F the data will be calculated with the wrong unit.